In [1]:
import pandas as pd

In [2]:
w_dummy_results = pd.read_csv('../docs_&_notes/first_ann_run_results_wo_dummy_cols.csv', delimiter=';')

In [3]:
w_dummy_results.head()

loss_function optimizer  \
0  binary_crossentropy     nadam   
1  binary_crossentropy     nadam   
2  binary_crossentropy     nadam   
3  binary_crossentropy     nadam   
4  binary_crossentropy     nadam   

                                      columns_used  batch_size  num_epochs  \
0  elo_diff, color, start_time, game_time, weekday           8          50   
1  elo_diff, color, start_time, game_time, weekday           8         100   
2  elo_diff, color, start_time, game_time, weekday           8         200   
3  elo_diff, color, start_time, game_time, weekday          20          50   
4  elo_diff, color, start_time, game_time, weekday          20         100   

   confusion_matrix  accuracy  
0  [40,  8][22, 30]        70  
1  [38, 10][20, 32]        70  
2  [35, 13][19, 33]        68  
3  [38, 10][19, 33]        71  
4  [38, 10][21, 31]        69

In [4]:
w_dummy_results.describe()

batch_size   num_epochs     accuracy
count  2641.000000  2641.000000  2641.000000
mean     38.711094   119.424460    68.829610
std      31.513613    62.684652     2.535903
min       8.000000    50.000000    60.000000
25%       8.000000    50.000000    67.000000
50%      20.000000   100.000000    69.000000
75%      44.000000   200.000000    71.000000
max      92.000000   200.000000    75.000000

In [5]:
w_dummy_results.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2641 entries, 0 to 2640
Data columns (total 7 columns):
loss_function       2641 non-null object
optimizer           2641 non-null object
columns_used        2641 non-null object
batch_size          2641 non-null int64
num_epochs          2641 non-null int64
confusion_matrix    2641 non-null object
accuracy            2641 non-null int64
dtypes: int64(3), object(4)
memory usage: 144.5+ KB


### Let's see which columns give the best average accuracy 

In [6]:
grouped_by_cols = w_dummy_results.groupby(by='columns_used').mean()
grouped_by_cols.sort_values('accuracy', ascending=False)

batch_size  num_epochs  \
columns_used                                                              
elo_diff, start_time, wkdy                        39.012048  119.879518   
elo_diff, start_time                              38.690909  119.393939   
elo_diff, start_time, game_time, weekday          38.690909  119.393939   
elo_diff, start_time, game_time                   38.690909  119.393939   
elo_diff, gt, wkdy                                38.690909  119.393939   
elo_diff, color, start_time, game_time, weekday   38.690909  119.393939   
elo_diff, wkdy                                    38.690909  119.393939   
elo_diff, gt                                      38.690909  119.393939   
elo_diff                                          38.690909  119.393939   
elo_diff, color, start_time, wkdy                 38.690909  119.393939   
elo_diff, color, start_time, game_time            38.690909  119.393939   
elo_diff, color, start_time                       38.690909  119.393939   
elo_diff, color, gt                               38.690909  119.393939   
elo_diff, color, gt, wkdy                         38.690909  119.393939   
elo_diff, color, wkdy                             38.690909  119.393939   
elo_diff, color                                   38.690909  119.393939   

                                                  accuracy  
columns_used                                                
elo_diff, start_time, wkdy                       71.060241  
elo_diff, start_time                             70.333333  
elo_diff, start_time, game_time, weekday         70.254545  
elo_diff, start_time, game_time                  70.024242  
elo_diff, gt, wkdy                               69.890909  
elo_diff, color, start_time, game_time, weekday  69.806061  
elo_diff, wkdy                                   69.587879  
elo_diff, gt                                     69.381818  
elo_diff                                         69.193939  
elo_diff, color, start_time, wkdy                69.157576  
elo_diff, color, start_time, game_time           69.127273  
elo_diff, color, start_time                      68.157576  
elo_diff, color, gt                              66.757576  
elo_diff, color, gt, wkdy                        66.448485  
elo_diff, color, wkdy                            66.060606  
elo_diff, color                                  66.018182

### Winners and Loser (+Fringers?):
#### Winners (Greater than 70%):
- elo_diff, start_time, weekday = 71.1%
- elo_diff, start_time = 70.3%
- elo_diff, start_time, game_time, weekday = 70.8%
- elo_diff, start_time, game_time = 70.3%

#### Losers (Less than 2/3 (66.67%)):
- elo_diff, color = 66.0%
- elo_diff, color, wkdy = 66.1%
- elo_diff, color, gt, wkdy = 66.4%
- elo_diff, color, gt = 66.8%

#### Winners on the fringe (Above 69%, less than 70%):
- elo_diff, gt, wkdy = 69.9%
- elo_diff, color, start_time, game_time, weekday = 69.8%
- elo_diff, wkdy = 69.6%
- elo_diff, gt = 69.4%
- elo_diff = 69.2%
- elo_diff, color, start_time, wkdy = 69.2%
- elo_diff, color, start_time, game_time = 69.1%

#### Losers on the fringe (Above 66.67%, less than 68%):
__NONE__

#### In-Betweeners (Between 68% and 69%):
- elo_diff, color, start_time = 68.2%

### Real Test! Group columns by the column name, loss function, and optimizer

In [7]:
grouped_by_cols = w_dummy_results.groupby(by=['columns_used', 'loss_function', 'optimizer']).mean()
grouped_df = grouped_by_cols.sort_values(by='accuracy', ascending=False).reset_index()

In [12]:
grouped_df[grouped_df['accuracy']>71.5]

columns_used        loss_function optimizer  batch_size  \
0  elo_diff, start_time, wkdy                  mae  adadelta        18.0   
1              elo_diff, wkdy  binary_crossentropy   adagrad        41.0   
2  elo_diff, start_time, wkdy                  mae      adam        41.0   
3              elo_diff, wkdy                  mse   adagrad        41.0   
4  elo_diff, start_time, wkdy                  mae     nadam        41.0   
5        elo_diff, start_time                  mae  adadelta        18.0   
6  elo_diff, start_time, wkdy                  mae   adagrad        41.0   
7  elo_diff, start_time, wkdy                  mae   rmsprop        41.0   
8                    elo_diff  binary_crossentropy   adagrad        41.0   
9                    elo_diff                  mse   adagrad        41.0   

   num_epochs   accuracy  
0  141.666667  72.166667  
1  116.666667  72.000000  
2  116.666667  72.000000  
3  116.666667  71.916667  
4  116.666667  71.833333  
5  141.666667  71.666667  
6  116.666667  71.583333  
7  116.666667  71.583333  
8  116.666667  71.583333  
9  116.666667  71.583333

In [13]:
grouped_df[grouped_df['accuracy']<66]

columns_used        loss_function optimizer  batch_size  \
217  elo_diff, color, gt, wkdy                  mae     nadam        41.0   
218        elo_diff, color, gt                  mae      adam        41.0   
219        elo_diff, color, gt                  mae   rmsprop        41.0   
220            elo_diff, color                  mae     nadam        41.0   
221        elo_diff, color, gt                  mae  adadelta        18.0   
222      elo_diff, color, wkdy                  mae     nadam        41.0   
223      elo_diff, color, wkdy                  mse     nadam        41.0   
224  elo_diff, color, gt, wkdy                  mse     nadam        41.0   
225      elo_diff, color, wkdy                  mae   adagrad        41.0   
226            elo_diff, color                  mae  adadelta        18.0   
227  elo_diff, color, gt, wkdy                  mae   adagrad        41.0   
228  elo_diff, color, gt, wkdy                  mae      adam        41.0   
229      elo_diff, color, wkdy                  mae  adadelta        18.0   
230        elo_diff, color, gt                  mae   adagrad        41.0   
231  elo_diff, color, gt, wkdy                  mae   rmsprop        41.0   
232            elo_diff, color                  mse     nadam        41.0   
233      elo_diff, color, wkdy                  mae   rmsprop        41.0   
234            elo_diff, color                  mae      adam        41.0   
235            elo_diff, color  binary_crossentropy     nadam        41.0   
236      elo_diff, color, wkdy                  mae      adam        41.0   
237      elo_diff, color, wkdy  binary_crossentropy     nadam        41.0   
238            elo_diff, color                  mae   adagrad        41.0   
239            elo_diff, color                  mae   rmsprop        41.0   

     num_epochs   accuracy  
217  116.666667  65.916667  
218  116.666667  65.750000  
219  116.666667  65.750000  
220  116.666667  65.750000  
221  141.666667  65.666667  
222  116.666667  65.416667  
223  116.666667  65.333333  
224  116.666667  65.250000  
225  116.666667  65.250000  
226  141.666667  65.166667  
227  116.666667  65.083333  
228  116.666667  65.083333  
229  141.666667  64.833333  
230  116.666667  64.750000  
231  116.666667  64.750000  
232  116.666667  64.666667  
233  116.666667  64.500000  
234  116.666667  64.416667  
235  116.666667  64.333333  
236  116.666667  64.250000  
237  116.666667  63.916667  
238  116.666667  63.833333  
239  116.666667  63.500000

In [16]:
print('BY Column Name:\nLess than 67%:')
print(grouped_df[grouped_df['accuracy']<=67].columns_used.value_counts())
print('\n\nBY Column Name:\nGreater than 71%:')
print(grouped_df[grouped_df['accuracy']>=71].columns_used.value_counts())
print('\n\nBY Loss Function:\nLess than 67%:')
print(grouped_df[grouped_df['accuracy']<=67].loss_function.value_counts())
print('\n\nBY Loss Function:\nGreater than 71%:')
print(grouped_df[grouped_df['accuracy']>=71].loss_function.value_counts())
print('\n\nBY Optimizer:\nLess than 67%:')
print(grouped_df[grouped_df['accuracy']<=67].optimizer.value_counts())
print('\n\nBY Optimizer:\nGreater than 71%:')
print(grouped_df[grouped_df['accuracy']>=71].optimizer.value_counts())

BY Column Name:
Less than 67%:
elo_diff, color, gt          10
elo_diff, color, gt, wkdy    10
elo_diff, color, wkdy        10
elo_diff, color               9
elo_diff                      2
elo_diff, gt                  2
elo_diff, wkdy                1
Name: columns_used, dtype: int64


BY Column Name:
Greater than 71%:
elo_diff, start_time, wkdy                         9
elo_diff                                           3
elo_diff, wkdy                                     3
elo_diff, start_time                               2
elo_diff, gt                                       2
elo_diff, color, start_time, game_time, weekday    1
Name: columns_used, dtype: int64


BY Loss Function:
Less than 67%:
mae                    24
mse                    12
binary_crossentropy     8
Name: loss_function, dtype: int64


BY Loss Function:
Greater than 71%:
mae                    7
mse                    7
binary_crossentropy    6
Name: loss_function, dtype: int64


BY Optimizer:
Less than 67%:
